File: 장승민/Java/210113_class2Practice2/src/com/jsm/carApp/customer/Customer.java
package com.jsm.carApp.customer;

import com.jsm.carApp.car.Car;

public class Customer {
	private String name;
	private String tel;
	private String address;
	private Car car;
	private static Customer inst;
	
	public static Customer getInstance(String name, String tel, String address, Car car) {
		if(inst==null) {
			inst=new Customer(name, tel, address, car);
		}
		return inst;
	}
	
	
	public Customer(String name, String tel, String address, Car car) {
		super();
		this.name = name;
		this.tel = tel;
		this.address = address;
		this.car = car;
	}




	public void printCustomerInfo() {
		System.out.println("고객명 :"+name);
		System.out.println("전화번호 :"+tel);
		System.out.println("주소 :"+address);
		car.printCarInfo();
	}
	
}

File: 황창식/C_sharp/Class_practice_20210215/practice_2/Customer.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace practice_2
{
    class Customer
    {
        private string name;
        private string tel;
        private string address;
        private Car car;

        public Customer(string name, string tel, string address, Car car)
        {
            this.name = name;
            this.tel = tel;
            this.address = address;
            this.car = car;
        }

        public string Name { get => name; set => name = value; }
        public string Tel { get => tel; set => tel = value; }
        public string Address { get => address; set => address = value; }
        internal Car Car { get => car; set => car = value; }

        public void printCustomerInfo()
        {
            Console.WriteLine("----- 고객/차량 정보 -----");
            Console.WriteLine("고객 이름 : " + name);
            Console.WriteLine("고객 전화 : " + tel);
            Console.WriteLine("고객 주소 : " + address);
            car.printCarInfo();
        }
    }
}
